docs/book/src/topics/ipv6-enabled-cluster.md

```diff
@@ -0,0 +1,368 @@
+
+```yaml
+spec:
+  network:
+    vpc:
+      ipv6: {}
+```
+
+**Note:** CAPA, by default, will provision a dualstack infrastructure (i.e. dualstack VPC and subnets). However, your Kubernetes cluster can be configured as either IPv6-only or dualstack depending on your pod/service CIDR configuration.
+
+## IPv6 CIDR Allocations
+
+CAPA supports various methods to allocate an IPv6 CIDR to the cluster VPC.
+
+### AWS-assigned IPv6 VPC CIDR
+
+To request AWS to automatically assign an IPv6 CIDR from an AWS defined address pool, use the following setting:
+
+```yaml
+spec:
+  network:
+    vpc:
+      ipv6: {}
+```
+
+By default, Amazon provides one fixed size (`/56`) IPv6 CIDR block to a VPC.
+
+### Bring-your-own IPv6 VPC CIDR (EC2)
+
+If you own an IPv6 address space, you can import it into AWS EC2 IPv6 address pool (See [guide](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html#byoip-requirements)). After importing it, you can assign /56 ranges from the space to individual VPCs in the same account.
+
+To define your own IPv6 address pool and CIDR set the following values:
+
+```yaml
+spec:
+  network:
+    vpc:
+      ipv6:
+        poolId: pool-id
+        cidrBlock: "2009:1234:ff00::/56"
+```
+
+### Bring-your-own IPv6 VPC CIDR via VPC Address Manager (VPC IPAM)
+
+If you want to allocate an IPv6 CIDR to the VPC from an existing VPC IPAM pool, define the pool ID and a prefix length as follows:
+
+```yaml
+spec:
+  network:
+    vpc:
+      ipv6:
+        ipamPool:
+          id: ipam-pool-id
+          netmaskLength: 56
+```
+
+By default, if you omit `netmaskLength`, CAPA will set it to the default `56`.
+
+### Bring-your-own IPv6 VPC
+
+If you have an existing dualstack VPC that you would like to use, you must explicitly provide the IPv6 CIDR block and egress-only internet gateway ID specs:
+
+```yaml
+spec:
+  network:
+    vpc:
+      id: vpc-1234567890abcdefg
+      cidrBlock: 10.0.0.0/16
+      ipv6:
+        cidrBlock: "2001:1234:ff00::/56"
+        egressOnlyInternetGatewayId: eigw-1234567890abcdefg
+```
+
+This has to be done to explicitly express the user intention to use the IPv6 capabilities of the VPC.
+
+## Creating IPv6 EKS-managed Clusters
+
+To quickly deploy an IPv6 EKS cluster, use the [IPv6 EKS cluster template](https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-aws/refs/heads/main/templates/cluster-template-eks-ipv6.yaml).
+
+<aside class="note warning">
+
+<h1>Warning</h1>
+
+EKS currently only supports IPv6-only clusters (not dualstack). You can't define custom Pod CIDRs on EKS with IPv6. EKS automatically assigns an address range from a unique local address range of `fc00::/7`.
```

Comment:
if I followed correctly the verbiage standard around that feature, it uses "IPv6 only", "dual-stack" (or "dual stack") for the feature references, and `dualstack` when referring to the option/attribute's value. Is that makes sense?
```suggestion
EKS currently only supports IPv6 only clusters (not dual-stack). You can't define custom Pod CIDRs on EKS with IPv6. EKS automatically assigns an address range from a unique local address range of `fc00::/7`.
```

[1] https://docs.aws.amazon.com/eks/latest/best-practices/ipv6.html
[2] https://docs.aws.amazon.com/vpc/latest/userguide/aws-ipv6-support.html